src/mixins/nr-affiliation-mixin.ts

```diff
@@ -141,7 +145,7 @@ export class NrAffiliationMixin extends Mixins(CommonMixin) {
     let businessRequest = {} as BusinessRequest
     const nrNumber = nr.nrNum
     if (!this.isFirm(nr)) {
-      if (nr.request_action_cd === 'AML') {
+      if (nr.request_action_cd === NrRequestActionCodes.AMALGAMATE) {
```

Comment:
Could you use the store getters here instead?

Reply:
Sev, I've tested this, I can use the `this.getNr` store getter here. It'll be `this.getNr.request_action_cd`.

But I've been thinking, since we're using the function and we're taking the nr parameter, how beneficial would it be to use the store getter in this case?

Reply:
Poke.

Reply:
Or maybe you're thinking to try removing the parameter altogether and using the store getter instead?

Reply:
I was thinking you'd use isAmalgamation/isContinuationIn/isNewBusiness ...?

Reply:
OH, these doesn't work as these getters are catching what the action item is. They don't actually check what the NR was.

Reply:
Aside from the possible code duplication (2 ways to do the same thing) in this module, it bothers me a bit that, here, you're checking a field in a formal parameter, while in the code below, the getters look at the global state.

But maybe these methods are used in totally different flows with different store states? If so, that's really not clear in here. But how much do we care about this? Should understanding this complexity be a rite of passsage for new developers? Dunno.

Reply:
True, I totally get where you're coming from.

From my understanding, the store getters (isAmalgamation/isContinuationIn/isNewBusiness) are checking when the action is selected from the dropdown menu.

This here on the other hand, is checking when we open an NR to see what was it's action. So, two completely different things.

I agree, this can be a bit misleading as one might think that for example `isAmalgamation` will return true when we open an NR amalgamation but that isn't the case. It will return true when amalgamation is selected in the dropdown menu.

How would you like to proceed Sev? One suggestion can be renaming those getters so that it's clearer on what they're checking for. Thoughts?

Reply:
Just leave it. We've spent enough time on this today.

Reply:
Your new code is consistent with the previous code. That's the most important thing here.